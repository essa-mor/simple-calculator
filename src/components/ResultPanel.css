.result_label{
	height: 16px;
	color: white;
	text-align: right;
	display: block;
	font-family: Arial;
	font-size: 13px;
	background-color: rgb(67, 52, 55);
	padding-top: 5px;
	padding-right: 8px;
  }
  
  .result_input{
	text-align: right;
	background-color: rgb(67, 52, 55);
	color: white;
	border: none;
	height: 25px;
	width: 159px;
	padding-right: 5px;
	font-weight: bold;
  }
  
  