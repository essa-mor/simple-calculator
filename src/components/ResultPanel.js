import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import './ResultPanel.css';

class ResultPanel extends PureComponent {
	constructor(props) {
		super(props);
		this.textInput = React.createRef();
	}

	componentDidMount() {
		this.textInput.current.focus();
	}

	render() {
		const { summary, result, className } = this.props;
		return (<div className={className}>
			<div className="box  ellipsis  reverse-ellipsis"><span className="result_label">{summary}</span></div>
			<input type="text" ref={this.textInput} value={isFinite(result) ? result : 'Can\'t devide by zero'} className="result_input" readOnly />
		</div>);
	}
}

ResultPanel.propTypes = {
	summary: PropTypes.string, 
	result: PropTypes.string, 
	className: PropTypes.string
};

export default ResultPanel;