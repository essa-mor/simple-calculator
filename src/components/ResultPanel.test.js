import React from 'react';
import renderer from 'react-test-renderer';

import ResultPanel from './ResultPanel';

const focusMock = {
	createNodeMock: element => {
		if (element.type === 'input') {
			// mock a focus functiona
			return {
				focus: () => {}
			};
		}
		return null;
	}
};

describe('ResultPanel test', () =>{
	it('Renders properly', () => {
		const tree = renderer.create(<ResultPanel summary="1 +" result="1" className="mycalss" />, focusMock).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('Renders properly - can\'t devide by zero', () => {
		const tree = renderer.create(<ResultPanel summary="1 +" result="Infinity" className="mycalss" />, focusMock).toJSON();
		expect(tree).toMatchSnapshot();
	});
});