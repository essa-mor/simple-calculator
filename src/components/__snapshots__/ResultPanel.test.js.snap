// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultPanel test Renders properly - can't devide by zero 1`] = `
<div
  className="mycalss"
>
  <div
    className="box  ellipsis  reverse-ellipsis"
  >
    <span
      className="result_label"
    >
      1 +
    </span>
  </div>
  <input
    className="result_input"
    readOnly={true}
    type="text"
    value="Can't devide by zero"
  />
</div>
`;

exports[`ResultPanel test Renders properly 1`] = `
<div
  className="mycalss"
>
  <div
    className="box  ellipsis  reverse-ellipsis"
  >
    <span
      className="result_label"
    >
      1 +
    </span>
  </div>
  <input
    className="result_input"
    readOnly={true}
    type="text"
    value="1"
  />
</div>
`;
